silo/tests/test_siloview.py

```diff
@@ -189,3 +191,305 @@ def test_data_silo_sort(self):
             current_rank = int(d['rank'])
             self.assertTrue(last_rank < current_rank)
             last_rank = current_rank
+
+
+class MergeTwoSilosTest(TestCase):
+    def setUp(self):
+        self.mapping_data = """{
+                "0": {
+                    "left_table_cols": ["number"],
+                    "right_table_col": "number",
+                    "merge_type": ""
+                },
+                "left_unmapped_cols": ["first name"],
+                "right_unmapped_cols": ["last name"]
+            }"""
+
+    def tearDown(self):
+        LabelValueStore.objects.delete()
+
+    def test_merge_silo(self):
+        user = factories.User(username='test_user')
+        left_silo = factories.Silo(owner=user, name='left_silo', public=True)
+        left_silo_r = factories.LabelValueStore(silo_id=left_silo.pk)
+        left_silo_r['number'] = 1
+        left_silo_r['first name'] = 'Bob'
+        left_silo_r.save()
+        factories.UniqueFields(silo=left_silo, name='number')
+
+        right_silo = factories.Silo(owner=user, name='right_silo', public=True)
+        right_silo_r = factories.LabelValueStore(silo_id=right_silo.pk)
+        right_silo_r['number'] = 1
+        right_silo_r['last name'] = 'Marley'
+        right_silo_r.save()
+        factories.UniqueFields(silo=right_silo, name='number')
```

Comment:
You're creating a left and right silo all the time. Don't you think that it's better to create a function and call it instead of copying and pasting?

Reply:
In some test cases one of fields or unique fields, or entire table missing to test it. Look at the next test cases. it's not completely same.

Reply:
But you're using quite often the same code. If you create a function to do this part, you don't need to call it in the test that you don't need it.

Reply:
I saw this code repeating seven times:
```
user = factories.User(username='test_user')
        left_silo = factories.Silo(owner=user, name='left_silo', public=True)
        left_silo_r = factories.LabelValueStore(silo_id=left_silo.pk)
        left_silo_r['number'] = 1
        left_silo_r['first name'] = 'Bob'
        left_silo_r.save()

        right_silo = factories.Silo(owner=user, name='right_silo', public=True)
        right_silo_r = factories.LabelValueStore(silo_id=right_silo.pk)
        right_silo_r['number'] = 1
        right_silo_r['last name'] = 'Marley'
        right_silo_r.save()
```